systemvm/debian/root/health_checks/iptables_check.py

```diff
@@ -38,10 +40,13 @@ def main():
                      ["POSTROUTING", "--to-source"]]:
             entriesExpected.append([algo[0], srcIpText, srcPortText, algo[1] + " " + dstText])
 
-        pout = Popen("iptables-save | grep " + destIp, shell=True, stdout=PIPE)
+        fetchIpTableEntriesCmd = "iptables-save | grep " + destIp
+        pout = Popen(fetchIpTableEntriesCmd, shell=True, stdout=PIPE)
         if pout.wait() != 0:
-            print "Unable to execute iptables-save command for fetching rules"
-            exit(1)
+            failedCheck = True
+            failureMessage = failureMessage + "Unable to execute iptables-save command " \
+                                              "for fetching rules by " + fetchIpTableEntriesCmd + "\n"
+            continue
```

Comment:
so we catch and remember an error but do not log it and than later log only the last one, again?

Reply:
We are concatenating all errors here like previous comment.